File: examples/generate_simple/generate_simple.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;
using System.Threading.Tasks; 

// Namespace of the generated Yarrow c# classes
using YarrowLib.Domain;

namespace Generate_class_framework
{
    class Program
    {
        static void Main(string[] args)
        {
            Info info = new Info()
            {
                Source = "Demo",
                Date_created = DateTime.Now
            };

            YarrowDataset_pydantic yarrowDataset = new YarrowDataset_pydantic()
            {
                Info = info,
                Images = new List<Image_pydantic>() 
                {
                    new Image_pydantic()
                        {
                            Id = "1",
                            Width = 1000,
                            Height = 1000,
                            File_name = "test_image.jpg",
                            Date_captured = DateTime.Now,
                            Azure_url = "https://picsum.photos/200/300"
                        }
                }
            };

            try
            {
                string output = JsonConvert.SerializeObject(yarrowDataset);
                Console.WriteLine($"Serialization succeded : \n {output}");

                File.WriteAllText("examples/generate_simple/example_simple.json", output);
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
            }
        }
    }
}

File: examples/generate_simple/generate_simple.py
import json
from datetime import datetime

import yarrow

if __name__ == "__main__":
    info = yarrow.Info(source="Demo", date_created=datetime.now())

    image = yarrow.Image(
        width=1000,
        height=1000,
        file_name="test_image.jpg",
        date_captured=datetime.now(),
        azure_url="https://picsum.photos/200/300",
    )

    yarrowset = yarrow.YarrowDataset(info=info, images=[image])

    yarrowset.pydantic().save_to_file(
        "examples/generate_simple/example_simple.yarrow.json",
    )
